### PR TITLE
 [10.0][IMP] shopinvader_locomotive_reset_password: Add password expiration management

### PR DESCRIPTION
**Pull request by rousseldenis**
* [x] Improve wizard to allow partner_ids from create
* [x] Adds password expiration and reset mechanism

### CODE REVIEW

**Comment by rousseldenis** [comment]
@Cedric-Pigeon @sebastienbeau @lmignon @simahawk

File: shopinvader_locomotive_reset_password/data/ir_cron.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2020 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo noupdate="1">
+
+    <record forcecreate="True" id="ir_cron_reset_expired_password" model="ir.cron">
+        <field name="name">Shopinvader - Reset Expired Passwords</field>
+        <field eval="True" name="active"/>
```
Comment (Cedric-Pigeon) [inline shopinvader_locomotive_reset_password/data/ir_cron.xml]: would set active False by default as this process is not mandatory

File: shopinvader_locomotive_reset_password/models/shopinvader_backend.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+class ShopinvaderBackend(models.Model):
+
+    _inherit = "shopinvader.backend"
+
+    password_validity = fields.Integer(
+        default=0,
+        string="Password Validity (days)",
+        help="Define here the user password validity (in days) on Website. "
+        "If the validity is passed, an email would be sent to reset it."
+        "To deactivate it, set it to 0.",
```
Comment (Cedric-Pigeon) [inline shopinvader_locomotive_reset_password/models/shopinvader_backend.py:20]: I would set -1 to inactivate. some process on shopinvader already use this convention.

**Review by Cedric-Pigeon — CHANGES_REQUESTED** [review]

File: shopinvader_locomotive_reset_password/models/shopinvader_backend.py (hunk 2 above)
Comment (rousseldenis) [reply shopinvader_locomotive_reset_password/models/shopinvader_backend.py:20]: euh... where ? And an integer cannot be negative.

File: shopinvader_locomotive_reset_password/tests/test_password_expired_reset.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.addons.queue_job.job import Job
+from odoo.addons.shopinvader.tests.common import CommonCase
+
+
+class TestShopinvaderResetExpiredPassword(CommonCase):
+    def setUp(self):
+        super(TestShopinvaderResetExpiredPassword, self).setUp()
+        self.backend.password_validity = 10
```
Comment (simahawk) [inline shopinvader_locomotive_reset_password/tests/test_password_expired_reset.py]: can you use setUpClass?

**Review by simahawk — APPROVED** [review]

File: shopinvader_locomotive_reset_password/models/shopinvader_backend.py (hunk 2 above)
Comment (Cedric-Pigeon) [reply shopinvader_locomotive_reset_password/models/shopinvader_backend.py:20]: see there for e.g. https://github.com/shopinvader/odoo-shopinvader/blob/10.0/shopinvader/models/shopinvader_notification.py#L69

Comment (rousseldenis) [reply shopinvader_locomotive_reset_password/models/shopinvader_backend.py:20]: Indeed

Comment (rousseldenis) [reply shopinvader_locomotive_reset_password/models/shopinvader_backend.py:20]: But does it have sense of 0 days validity ?

Comment (Cedric-Pigeon) [reply shopinvader_locomotive_reset_password/models/shopinvader_backend.py:20]: no indeed...

**Review by Cedric-Pigeon — APPROVED** [review]

**Review by sebastienbeau — COMMENTED** [review]
Hi looking at the code
If the password arrive at expiration an email will be send to the customer but if the customer do not care he can still use it's old password to connect.
So the process is not a blocking/mandatory process but just an volontary one.
Maybe it worst to add some comment in the readme about this, what do you think?

On my side I will soon propose an alternative module tat allow in some special case to do the authentication throught a odoo rest service (the logic of auth will be done in odoo and loco will only use it). This approche is interesting when you need some advanced custom behaviour

**Comment by sebastienbeau** [comment]
@Cedric-Pigeon @rousseldenis check my comment

**Comment by rousseldenis** [comment]
> Hi looking at the code
> If the password arrive at expiration an email will be send to the customer but if the customer do not care he can still use it's old password to connect.
> So the process is not a blocking/mandatory process but just an volontary one.
> Maybe it worst to add some comment in the readme about this, what do you think?
> 
> On my side I will soon propose an alternative module tat allow in some special case to do the authentication throught a odoo rest service (the logic of auth will be done in odoo and loco will only use it). This approche is interesting when you need some advanced custom behaviour

Ok, will change README

**Comment by rousseldenis** [comment]
Done

**Review by sebastienbeau — APPROVED** [review]
Ok for me

**Pull request merged**